students/TinaB/lesson04/mailroom_part2.py

```diff
@@ -0,0 +1,231 @@
+    user_input = input("\nLookup donor by: \n"
+                       "\t1. Firstname: \n"
+                       "\t2. Lastname:\n"
+                       "Please enter 1 or 2 >>  ")
+
+    donor_return = []
+    donor_check = 0
+    if int(user_input) == 1:
+        fname = input(
+            "Type the first name of the donor you are looking for:  ")
+    else:
+        lname = input("Type the last name of the donor you are looking for:  ")
+
+    for donor in donors_list_dictionary:
+        # print(donor["firstname"].lower())
+        if int(user_input) == 1:
+            if fname.lower() in donor["firstname"].lower():
+                donor_check = input("Is this the donor you are looking for: {} {}?  Please enter yes or no:  ".format(
+                    donor["firstname"], donor["lastname"]))
+                if donor_check == "yes":
+                    donor_return.append(donor["firstname"])
+                    donor_return.append(donor["lastname"])
+                    donor_check = 1
+                    break
+            else: # firstname is not in list
+                print("HERE!")
+                first_name = fname
+                last_name = input(
+                    "No donor with that first name found in donor list. Please provide a last name to add donor to list: ")
+                donor_return.append(first_name)
+                donor_return.append(last_name)
+                break
+
+        else:  # user input = 2
+            if lname.lower() in donor["lastname"].lower():
+                donor_check = input(
+                    "Is this the donor you are looking for:  {} {}?   Please enter yes or no:  ".format(donor["firstname"], donor["lastname"]))
+                if donor_check == "yes":
+                    donor_return.append(donor["firstname"])
+                    donor_return.append(donor["lastname"])
+                    donor_check = 1
+                    break
+                else:
+                    last_name = lname
+                    first_name = input(
+                        "No donor with that last name found in donor list. Please provide a first name for addition to donor list")
+                    donor_return.append(first_name)
+                    donor_return.append(last_name)
+                    break
+
+    return donor_return, donor_check
+
+
+def send_single_thank_you():
+    """ function for sending thank you message"""
+    donor_name, donor_check = get_name_input()
+    donor_amount = float(input('Please enter a donation amount for {} {}: '.format(
+        donor_name[0], donor_name[1])))
+
+    # search through donors with returned name - 1 means name is in list, 0 means not in list
+    if donor_check == 1:
+        for donor in donors_list_dictionary:
+            if donor["firstname"] == donor_name[0] and donor["lastname"] == donor_name[1]:
+                donor["donations"].append(donor_amount)
+                donor["donate_times"]+=1
```

Comment:
`donor["donate_times"]` is a list, so you'd need to access the item that you want to increment, but you don't appear to be using the "donate_times" value anywhere in your code so I would just get rid of it and use `len("donations")` where you need the total number of donations